src/main/java/org/openstreetmap/atlas/geography/Polygon.java

```diff
@@ -248,6 +257,75 @@ public Location interiorCenter()
         return new JtsLocationConverter().backwardConvert(point.getCoordinate());
     }
 
+    /**
+     * @param expectedNumberOfSides
+     *            Expected number of sides
+     * @param threshold
+     *            {@link Angle} threshold that decides whether a {@link Heading} difference between
+     *            segments should be counted towards heading change count or not
+     * @return true if this {@link Polygon} has approximately n sides while ignoring {@link Heading}
+     *         differences between inner segments that are below given threshold.
+     */
+    public boolean isApproximatelyNSided(final int expectedNumberOfSides, final Angle threshold)
+    {
+        // Ignore if polygon doesn't have enough inner shape points
+        if (expectedNumberOfSides < MINIMUM_N_FOR_SIDE_CALCULATION
+                || this.size() < expectedNumberOfSides)
+        {
+            return false;
+        }
+
+        // An N sided shape should have (n-1) heading changes
+        final int expectedHeadingChangeCount = expectedNumberOfSides - 1;
+
+        // Fetch segments and count them
+        final List<Segment> segments = this.segments();
+        final int segmentSize = segments.size();
+
+        // Index to keep track of segment to work on
+        int segmentIndex = 0;
+
+        // Keep track of heading changes
+        int headingChangeCount = 0;
+
+        // Find initial heading
```

Comment:
Is this a candidate for a helper method (not sure where it'd go, maybe a static method in Segment?)?  Finding the initial heading from a list of segments?

Reply:
I'd note this and extract it out when there are more clients needing this functionality.

Reply:
👍